### DISCUSSION TITLE
Disable OSD Audio

### DISCUSSION THREAD

**Discussion by chipsn** [MonitorControl/MonitorControl#711] [Q&A] [answered]
How can i disable monitor OSD for audio and only see macOS OSD (now i always see both or the monitor)?

**Comment by waydabber** [comment, marked as answer]
You can disable macOS OSD (so you see only the display's if your display shows an OSD for volume). To do this:

<img width="798" alt="Screen Shot 2021-10-13 at 10 36 13" src="https://user-images.githubusercontent.com/37590873/137097785-25162455-1617-4b32-80ac-000163c4f245.png">

You cannot disable the display's OSD. Version 2.1.0 had a setting for this but mostly it did not work (the OSD still flickered up) and had serious side effects for some displays (the OSD was completely disabled permanently and a factory reset was needed without an OSD...) so this feature has been removed.

**Reply by chipsn** [reply]
This worked for me and my LG monitor in 2.1.0, but now i have the monitor OSD flickering!

**Reply by chipsn** [reply]
Couldn't it also be an option at own risk (as in the advanced monitor settings)?!

**Reply by waydabber** [reply]
I understand the frustration. LG displays are notorious about showing the unwanted volume OSD (among other things - even better is the blinding full gray background upon sleep or the super bright blinking LED during sleep on some models that can light up the whole room at night). :) For most LG displays however this did not work properly (the OSD still showed up briefly and also volume control was extremely slow) while for some it might have worked.

The flickering may happen if you use the volume and brightness keys in tandem as changing the brightness also seems to hide the volume OSD on some LG models after the OSD was shown briefly .

I advise you to disable the macOS OSD if you want to avoid having a double OSD. We do not plan to add this option back even as an advanced option due to the possible harm that may come to some other displays with the disable OSD command. You can however build a custom version with this feature added (just look at the relevant code segment in v2.1.0 and apply it to v4.0.0).

**Reply by waydabber** [reply]
@chipsn - can you tell me exactly what model of LG display do you have?

On My LG display if I issue some neutral command (like set the current brightness again) after changing the volume, the same thing happens as when I'd issue the problematic OSD disable command (the volume OSD flickers in briefly but disappears instantly when the second command is issued). Since sending some neutral should be generally harmless (unlike sending the OSD disable command) I could add something like this as an advanced option and that would probably solve the issue for you.

**Reply by CtrlAltSudo** [reply]
I have the LG Gl850 and I'm experiencing the same symptoms when adjusting the volume... Perhaps @chipsn also as the GL850 or a similar model.
